include/rocm_smi/rocm_smi.h

```diff
@@ -274,6 +291,51 @@ rsmi_status_t rsmi_shut_down(void);
  */
 rsmi_status_t rsmi_num_monitor_devices(uint32_t *num_devices);
 
+/**
+ *  @brief Get the list of possible pci bandwidths that are available.
+ *
+ *  @details Given a device index @p dv_ind and a pointer to a to an
+ *  rsmi_pcie_bandwidth structure @p bandwidth, this function will fill in
+ *  @p bandwidth with the possible T/s values and associated number of lanes,
+ *  and indication of the current selection.
+ *
+ *  @param[in] dv_ind a device index
+ *
+ *  @param[inout] bandwidth a pointer to a caller provided rsmi_pcie_bandwidth
+ *  structure to which the frequency information will be written
+ *
+ *  @retval RSMI_STATUS_SUCCESS is returned upon successful call.
+ *
+ */
+rsmi_status_t
+rsmi_dev_pci_bandwidth_get(uint32_t dv_ind, rsmi_pcie_bandwidth *bandwidth);
+
+/**
+ *  @brief Control the set of allowed PCIe bandwidths that can be used.
+ *  
+ *  @details Given a device index @p dv_ind and a 64 bit bitmask @p bw_bitmask,
+ *  this function will limit the set of allowable bandwidths. If a bit in @p
+ *  bw_bitmask has a value of 1, then the frequency (as ordered in an
+ *  rsmi_frequencies returned by rsmi_dev_get_gpu_clk_freq()) corresponding
+ *  to that bit index will be allowed.
+ *  
+ *  This function will change the performance level to
+ *  ::RSMI_DEV_PERF_LEVEL_MANUAL in order to modify the set of allowable
+ *  band_widths. Caller will need to set to ::RSMI_DEV_PERF_LEVEL_AUTO in order
+ *  to get back to default state.
+ *  
+ *  All bits with indices greater than or equal to
+ *  rsmi_pcie_bandwidth.transfer_rate.num_supported will be ignored.
+ *
+ *  @param[in] dv_ind a device index
+ *
+ *  @param[in] bw_bitmask A bitmask indicating the indices of the
+ *  bandwidths that are to be enabled (1) and disabled (0). Only the lowest
+ *  rsmi_pcie_bandwidth.transfer_rate.num_supported bits of this mask are
+ *  relevant.
+ */
+rsmi_status_t rsmi_dev_pci_bandwidth_set(uint32_t dv_ind, uint64_t bw_bitmask);
```

Comment:
Is 64 always enough?  Will 64 be enough for other connectors and should the signature for all connectors be common?